src/API/Authentication.php

```diff
@@ -42,20 +37,16 @@ protected function setApiClient() {
     $this->apiClient = $client;
   }
 
-  public function get_oauth_client($client_secret, $redirect_uri, $extra_params = []) {
-
-    if (empty($this->client_id)) {
-      throw new ApiException('client_id was not set.');
-    } 
-
-    $extra_params['domain'] = $this->domain;
-    $extra_params['client_id'] = $this->client_id;
-    $extra_params['client_secret'] = $client_secret;
-    $extra_params['redirect_uri'] = $redirect_uri;
-
-    return new Oauth2Client($extra_params);
-  }
-
+  /**
+   * Builds and returns the `/authorize` url in order to initialize a new authN/authZ transaction
+   *
+   * @method get_authorize_link https://auth0.com/docs/api/authentication#!#get--authorize_db
+   * @param {String} response_type
+   * @param {String} redirect_uri
+   * @param {String} connection [optional]
+   * @param {String} state      [optional]
+   * @param {Object} aditional_params      [optional]
+   */
   public function get_authorize_link($response_type, $redirect_uri, $connection = null, $state = null, $aditional_params = []) {
```

Comment:
I prefer to have required options as parameters, everything else as an array but the method signature gets way to long. I like the idea of making everything an option but we should validate and document what are considered valid options and fail if they provide a non valid one to avoid mispelling of the keys: ie: `redirect_url` instead of `redirect_uri`